### PR TITLE
Add browser renderToStream

### PR DESCRIPTION
**Pull request by popeindustries**
(synthetic tests only - not tested in browser context)

### CODE REVIEW

File: src/browser-stream-template-renderer.js [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+        this.process = processor.getProcessor(
+          {
+            push(chunk) {
+              if (chunk === null) {
+                controller.close();
+                return false;
+              }
+
+              controller.enqueue(chunk);
```
Comment (robwebdev) [inline src/browser-stream-template-renderer.js]: For this to be read by a user constructed Response I believe the chunk needs to be encoded using [`TextEncoder.encode`](https://developer.mozilla.org/en-US/docs/Web/API/TextEncoder/encode) ([Example here](https://jakearchibald.com/2016/streams-ftw/#serving-a-string-slowly)).

**Comment by robwebdev** [comment]
Thanks so much for this!

I tested this in a browser context and hit 2 issues, the first is [this comment](https://github.com/popeindustries/lit-html-server/pull/8/files#r279147032)

The second is that the stream appears to be enqueuing duplicate chunks. I haven't had a chance to dig into the cause of that any further just yet though.

**Comment by popeindustries** [comment]
> The second is that the stream appears to be enqueuing duplicate chunks. I haven't had a chance to dig into the cause of that any further just yet though.

this should be fixed now

**Comment by robwebdev** [comment]
This is working perfectly in my test now 😎(Chrome)

**Pull request merged**